crates/tx-graph2/src/connectors/claim_contest_connector.rs

```diff
@@ -0,0 +1,439 @@
+
+/// Connector output between `Claim` and:
+/// 1. `UncontestedPayout`, and
+/// 2. `Contest`.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ClaimContestConnector {
+    network: Network,
+    n_of_n_pubkey: XOnlyPublicKey,
+    watchtower_pubkeys: Vec<XOnlyPublicKey>,
+    delta_contest: relative::LockTime,
+}
+
+impl ClaimContestConnector {
+    /// Creates a new connector.
+    pub const fn new(
+        network: Network,
+        n_of_n_pubkey: XOnlyPublicKey,
+        watchtower_pubkeys: Vec<XOnlyPublicKey>,
+        delta_contest: relative::LockTime,
+    ) -> Self {
+        Self {
+            network,
+            n_of_n_pubkey,
+            watchtower_pubkeys,
+            delta_contest,
+        }
+    }
+
+    /// Returns the number of watchtowers for the connector.
+    pub const fn n_watchtowers(&self) -> usize {
+        self.watchtower_pubkeys.len()
+    }
+
+    /// Returns the delta contest relative timelock of the connector.
+    ///
+    /// The sequence of the input and the global locktime must be
+    /// large enough to cover this value.
+    pub const fn delta_contest(&self) -> relative::LockTime {
+        self.delta_contest
+    }
+
+    /// Generates a vector of all leaf scripts of the connector.
+    pub fn leaf_scripts(&self) -> Vec<ScriptBuf> {
+        let mut scripts = Vec::new();
+
+        for watchtower_pubkey in &self.watchtower_pubkeys {
+            let script = bitcoin::script::Builder::new()
+                .push_slice(self.n_of_n_pubkey.serialize())
+                .push_opcode(opcodes::all::OP_CHECKSIGVERIFY)
+                .push_slice(watchtower_pubkey.serialize())
+                .push_opcode(opcodes::all::OP_CHECKSIG)
+                .into_script();
+            scripts.push(script);
+        }
+
+        let script = bitcoin::script::Builder::new()
+            .push_slice(self.n_of_n_pubkey.serialize())
+            .push_opcode(opcodes::all::OP_CHECKSIGVERIFY)
+            .push_sequence(self.delta_contest.to_sequence())
+            .push_opcode(opcodes::all::OP_CSV)
+            .into_script();
+        scripts.push(script);
+
+        scripts
+    }
+
+    /// Generates the bitcoin address and taproot spending info of the connector.
+    pub fn address_and_spend_info(&self) -> (bitcoin::Address, TaprootSpendInfo) {
+        let internal_key = self.n_of_n_pubkey;
+        let scripts = self.leaf_scripts();
+        create_taproot_addr(
+            &self.network,
+            SpendPath::Both {
+                internal_key,
+                scripts: scripts.as_slice(),
+            },
+        )
+        .expect("tap tree is valid")
+    }
+
+    /// Generates the txout of the connector.
+    pub fn tx_out(&self) -> TxOut {
+        let script_pubkey = self.address_and_spend_info().0.script_pubkey();
+        let dust = script_pubkey.minimal_non_dust();
+        // TODO: Replace magic number 3 with constant from contest transaction, once the code exists
```

Comment:
Each `TODO` must come with an assignee.

```suggestion
        // TODO: (@uncomputable) Replace magic number 3 with constant from contest transaction, once the code exists
```